### PR TITLE
[Transform] Add `delete_dest_index` parameter to the `Delete Transform API`

### PR DESCRIPTION
**Pull request by przemekwitek**
This PR introduces the new `delete_dest_index` parameter in the `Delete Transform API` request.
When set to `true`, the destination index is deleted together with the transform regardless of whether the destination index was created by the transform or by the user.
The default value is `false`, meaning that the destination index will not be deleted.

Special case: If destination index is an alias, it is **not** deleted and the following error message is emitted:
```
The provided expression [dest-index] matches an alias, specify the corresponding concrete indices instead.
```

Relates https://github.com/elastic/elasticsearch/issues/93794

### CODE REVIEW

**Review by hendrikmuhs — COMMENTED** [review]
This addition is nice.

Some thoughts:

- naming, see comment, it is called `dest` in the config, but `destination` here
- delete no matter what
  - what if `dest` is a data stream?[1]
  - what if `dest` is an alias?[1]
- what if the transform also created a pipeline? it would be good if we cover both[2]

[1] this might speak for deleting only auto-created indices. Otherwise you can run the GET index API, it will return something, even for aliases and data streams. The output will tell you, if you run it on an alias or data stream. This requires some interpretation of the result, which I hope isn't complex.
[2] the host risk score package afaik uses a pipeline

File: docs/reference/transform/apis/delete-transform.asciidoc [hunk 1]
```diff
@@ -36,6 +36,11 @@ include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=transform-id]
 current state. The default value is `false`, meaning that the {transform} must be
 `stopped` before it can be deleted.
 
+`delete_destination_index`::
```
[the remark this answers is not in the document]
Comment (przemekwitek) [reply docs/reference/transform/apis/delete-transform.asciidoc]: Done.

**Comment by przemekwitek** [comment]
> This addition is nice.
> 
> Some thoughts:
> 
> * naming, see comment, it is called `dest` in the config, but `destination` here

Done.

> * delete no matter what

Done.

>   * what if `dest` is a data stream?[1]
>   * what if `dest` is an alias?[1]

Good catch, these 2 are currently not covered.

> * what if the transform also created a pipeline?

You mean, the user had created the pipeline upfront? I don't think the transform creates a pipeline anywhere...

> it would be good if we cover both[2]
> 
> [1] this might speak for deleting only auto-created indices. Otherwise you can run the GET index API, it will return something, even for aliases and data streams. The output will tell you, if you run it on an alias or data stream. This requires some interpretation of the result, which I hope isn't complex.

Looking into this...

> [2] the host risk score package afaik uses a pipeline

Is it one of the Fleet packages?

**Comment by hendrikmuhs** [comment]
> You mean, the user had created the pipeline upfront? I don't think the transform creates a pipeline anywhere...

good point.

>>   [2] the host risk score package afaik uses a pipeline

>Is it one of the Fleet packages?

yes, I will try to find it. But as you noted the pipeline has been created by the [package](https://github.com/elastic/package-spec), so we can assume that it is cleaned up by the creator of the pipeline. The destination index however might have been created by transform, not by the package. I therefore take my request to add pipeline removal back, we don't need it.

**Comment by przemekwitek** [comment]
> * what if `dest` is a data stream?[1]

Then the destination index cannot be created on start.
I've added a test `TransformPivotRestSpecialCasesIT.testDataStreamUnsupportedAsDestIndex` for this case.

> * what if `dest` is an alias?[1]

Then it is not deleted, even if `delete_dest_index` is set to `true`.
I've added a test `TransformDeleteIT.testDeleteWithParamDoesNotDeleteAlias` for this case.

**Review by hendrikmuhs — APPROVED** [review]
LGTM

Thanks for taking the extra steps to set/fix task ids and timeouts.

**Pull request merged**